### PR TITLE
Added code to create a LeaseRequests collection in the WorkerProcess DB

### PR DESCRIPTION
**Pull request by nikhilgirraj**
Added a new LeaseRequests collection to CosmosDbLeaseStore init with -
1. Unique Key constraint on LeaseType (WorkerType) and InstanceId
2. CompositeKey over Priority and _ts (timestamp) for ORDER BY queries

### CODE REVIEW

File: src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs [hunk 1]
```diff
@@ -61,6 +67,22 @@ await _documentClient.CreateDatabaseIfNotExistsAsync(new Database {Id = _options
...
+        private async Task CreateLeaseRequestsCollection()
+        {
+            var collectionDefinition = new DocumentCollection
+            {
+                Id = _options.Value.RequestsCollection
+            };
+
+            collectionDefinition.PartitionKey.Paths.Add("/leaseType");
+            collectionDefinition.DefaultTimeToLive = 30;
+
```
Comment (faprovitolo) [inline src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:79]: Can we set an unique constraint on (WorkType,InstanceId) on LeaseRequests collection?

**Review by faprovitolo — CHANGES_REQUESTED** [review]

**Comment by nikhilgirraj** [comment]
As discussed with @faprovitolo, adding a Composite index and Unique key to the new collection.

File: src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs [hunk 2]
```diff
@@ -61,6 +67,36 @@ await _documentClient.CreateDatabaseIfNotExistsAsync(new Database {Id = _options
...
+            {
+                new CompositePath { Path = "/priority", Order = CompositePathSortOrder.Ascending },
+                new CompositePath { Path = "/_ts", Order = CompositePathSortOrder.Ascending }
+            };
+
+            collectionDefinition.UniqueKeyPolicy.UniqueKeys.Add(leaseConstraint);
+            collectionDefinition.IndexingPolicy.CompositeIndexes.Add(sortingIndex);
+
+            collectionDefinition.PartitionKey.Paths.Add("/leaseType");
+            collectionDefinition.DefaultTimeToLive = 30;
```
Comment (chomick) [inline src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:92]: Can we ensure that TTL is set per document instance, please? Not sure if we want to do it on collection level?

**Review by chomick — APPROVED** [review]

File: src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs (hunk 2 above)
Comment (nikhilgirraj) [reply src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:92]: @faprovitolo Can you add a TTL property to the LeaseRequest document in your task?

Comment (faprovitolo) [reply src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:92]: sure @nikhilgirraj but it should considered that, according to Microsoft docs:
"Time to Live on an item (set using ttl):

- This Property is applicable only if DefaultTimeToLive is present and it is not set to null for the parent container.

- If present, it overrides the DefaultTimeToLive value of the parent container."

I can possibly add TTL to each item setting it to 30 sec while setting defaulttimtolive to -1?

**Review by faprovitolo — APPROVED** [review]

File: src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs (hunk 2 above)
Comment (chomick) [reply src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:92]: The lease request delay is driven by config, isn't it? So it cannot be set here but if a value has to be set, it's fine if we default it to something - I assumed we can only set it when persisting the document.

**Pull request merged**